### PR TITLE
Initialize the docs site using Rust's mdBook

### PR DESCRIPTION
**Pull request by luxas**
Fixes: #4 

Works:
- [x] TOC auto-generation
- [x] LaTeX Math expressions
- [x] Syntax Highlighting
- [x] Highlighting of specific parts of a code file
- [x] Inclusion of a specific file
- [x] Footnotes
- [x] Showing the target URL of a link when printing

Yet to do:
- [x] Create a Dockerfile with Rust deps pre-installed
- [x] Integrate with GitHub Actions & Pages for a smooth CI/CD flow
- [x] Point `docs.racklet.io` to this repo's GH Actions

Does not work:
 - [ ] Abbreviations (as per https://michelf.ca/projects/php-markdown/extra/#abbr)
 - [ ] `<sup>` <-> 19^th^ <-> `19^th^`
 - [ ] `<sub>` <-> H~2~O <-> `H~2~O`
 - [ ] Ordering of footnotes (fix this somehow with a post-processor?)
 - [ ] Automatic inclusion of links into the references list at the end of the page (fix using a Markdown pre-processor?)

I could turn the book into a PDF with:
```
google-chrome-stable --headless --print-to-pdf=rust_book.pdf <built HTML file>
```
Possibly it is better to use Firefox in headless mode, IDK, but headless Chrome (or -ium) works at least.

EDIT: There is now extensive documentation of what features are included and how to use them in `docs/README.md`

### CODE REVIEW

**Review by twelho — APPROVED** [review]
LGTM :+1:
For the table of contents we could at some point also look at a generator that embeds it into the markdown so it is usable as-is when editing/viewing the files without mdBook. For now let's see how well `mdbook-toc` handles it.

File: docs/rfcs/0001-high-level-architecture.md [hunk 1]
```diff
@@ -333,3 +299,15 @@ For this project to be considered successful and graduated, we mandate the follo
...
+
+[^kubecloud]: "KubeCloud: A Small-Scale Tangible Cloud Computing Environment". Master's thesis in Computer Engineering at Aarhus University by Kasper Nissen and Martin Jensen. Published June 6th, 2016. [Download PDF here](https://github.com/KubeCloud/thesis/raw/master/master.pdf)
+
+[^trusted_firmware]: For example the [1st stage bootloader of the Raspberry Pi 4] is currently closed source software which we cannot audit or modify, and hence cannot use as a "complete end to end" hardware root of trust. However, such non-idealities don't stop us from getting as close as possible to full hardware root of trust, and more importantly, conceptually being consistent in the way we work with these SBCs and "normal" servers.
+
+[^security]: At least initially this does not mean that the system is 100% secure, there are both some practical limits[^trusted_firmware] and software/hardware features that need to be explored for improved security (e.g. [ARM TrustedFirmware]).
+
+[^Ethernet]: ["Ethernet - Wikipedia"][Ethernet]
```
Comment (twelho) [inline docs/rfcs/0001-high-level-architecture.md:313]: Don't make this a footnote

**Review by twelho — COMMENTED** [review]
LGTM so far, just a couple of nits.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	# constantly update this repo. If this breaks in the CI, it's not very critical, and we'll then adapt.
+	docker pull zenika/alpine-chrome:latest
+	# Refer to the great guide at https://github.com/Zenika/alpine-chrome#-the-best-with-seccomp
+	for file in $$(find docs/rfcs/*.md | sed "s|docs/||;s|.md||" | grep -v "0000-rfc-template" | grep -v "README"); do \
+		docker run $(RUN_ARGS) -e file=$$file --security-opt seccomp=$(shell pwd)/chrome-seccomp.json -v $(shell pwd):/build -w /build \
+		zenika/alpine-chrome:latest --disable-gpu --print-to-pdf-no-header --print-to-pdf=$(MDBOOK_OUTPUT)/$$file.pdf $(MDBOOK_OUTPUT)/$$file.html 1>/dev/null && \
+		echo "=> $(MDBOOK_OUTPUT)/$$file.pdf" && \
+		cp $(MDBOOK_OUTPUT)/$$file.pdf docs/$$file.pdf && \
+		echo "=> docs/$$file.pdf" \
+	; done
```
[the remark this answers is not in the document]
Comment (luxas) [reply Makefile:66]: This is needed; make does not understand that the line continues unless there is a `\`, and there must be a colon between the command and `done`. I'll keep this as is.

**Comment by luxas** [comment]
Thanks for the review @twelho, I will now go ahead and merge :tada: :ship:

**Pull request merged**